pinot-common/src/main/java/org/apache/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -103,6 +112,23 @@ public BrokerRequest compileToBrokerRequest(String expression)
 
       BrokerRequest brokerRequest = new BrokerRequest();
       rootNode.updateBrokerRequest(brokerRequest);
+      try {
+        PinotQuery pinotQuery = new PinotQuery();
+        rootNode.updatePinotQuery(pinotQuery);
+        if (VALIDATE_CONVERTER) {
```

Comment:
Might be better to write exhaustive tests, than to validate when serving real traffic.

Reply:
Every integration test goes through this test right? We have one test where we put all pinot queries in one file and compare the two parsers.

Reply:
pql2Compiler has a testConverter method that runs 1000+ queries. I just ran pinot tests and captured all the queries and put them in a file and used them as test case.